test/code_maat/app/grouper_test.clj

```diff
@@ -10,52 +10,106 @@
 "/some/path => G1
 /another/path => G2")
 
+(def ^:const multi-regexp-group-spec
+"^/some/path_\\w+_group1$ => G1
+^/another/path_\\w+_group2$ => G2")
+
+(def ^:const multi-mixed-group-spec
+"/some/path => G1
+^/another/path/\\.*$ => G2")
+
 (deftest parses-specification
   (testing "Single group"
-    (is (= (g/text->group-specification "/some/path => G1")
-           [{:path "/some/path" :name "G1"}])))
-  (testing "Multiple groups"
-    (is (= (g/text->group-specification multi-group-spec)
-           [{:path "/some/path" :name "G1"}
-            {:path "/another/path" :name "G2"}])))
-   (testing "No groups"
+    (let [result (g/text->group-specification "/some/path => G1")]
```

Comment:
The test cases are a bit too noisy. We need to simplify them by encapsulating a transformation that converts the regex object into a comparable string. Here are two functions for that purpose:

```
 (defn- comparable-group-spec
  "We need a way to compare the strings that make up the
   regex rather than the regex objects themselves."
  [s]
  (map #(update % :path str) s))

(defn- comparable-group-spec-for
  [text]
  (-> text
      g/text->group-specification
      comparable-group-spec))
```

With these two functions, we can simplify the tests themselves. Here's an example:

```
(deftest parses-specification
  (testing "Single group"
    (is (= (comparable-group-spec-for "/some/path => G1")
           (comparable-group-spec [{:path "^/some/path/"
                                    :name "G1"}]))))

  (testing "Multiple text groups"
    (is (= (comparable-group-spec-for multi-group-spec)
           (comparable-group-spec [{:path "^/some/path/"
                                    :name "G1"}
                                   {:path "^/another/path/"
                                    :name "G2"}]))))

... and so on...
````